src/dbnode/client/session.go

```diff
@@ -3143,10 +3151,12 @@ func (b *baseBlocksResult) segmentForBlock(seg *rpc.Segment) ts.Segment {
 
 func (b *baseBlocksResult) mergeReaders(start time.Time, blockSize time.Duration, readers []xio.SegmentReader) (encoding.Encoder, error) {
 	iter := b.multiReaderIteratorPool.Get()
+	iter.SetSchema(b.nsCtx.Schema)
```

Comment:
I'd be in favor of making `schema` be required by call to `Reset(...)` rather than being a separate method since it's required every time you start using the iterator anyhow.

Reply:
agree. this makes sense.